Mathlib/Tactic/Positivity/Basic.lean

```diff
@@ -447,3 +448,29 @@ def evalRatDen : PositivityExt where eval {u α} _ _ e := do
     assumeInstancesCommute
     pure $ .positive q(den_pos $a)
   | _, _ => throwError "not Rat.num"
+
+/-- Extension for `posPart`. `a⁺` is always nonegative, and positive if `a` is. -/
+@[positivity _⁺]
+def evalPosPart : PositivityExt where eval zα pα e := do
+  match e with
+  | ~q(@posPart _ $instαlat $instαgrp $a) =>
+    assertInstancesCommute
+    match ← catchNone (core zα pα a) with
```

Comment:
@eric-wieser, do you think this `catchNone` is a code smell? I would expect `core` not to fail but return `none`

Reply:
I think the whole error handling model of positivity is a code smell...